codalab/lib/codalab_manager.py

```diff
@@ -497,6 +517,69 @@ def client(self, address, is_cli=True):
     def cli_verbose(self):
         return self.config.get('cli', {}).get('verbose')
 
+    def get_access_token(self, address, auth_handler):
```

Comment:
I don't think you need to copy this code. Here's what I propose:

1) Get rid of the BundleClient login method entirely. It doesn't need to exist, and all it does is just call get_access_token.
2) Change authenticate to take in an address and auth_handler.
3) Use a lambda in both places where _authenticate needs to be called to pass in the address and auth_handler.

Am I missing something?